test/Hamiltonians.jl

```diff
@@ -1669,3 +1669,107 @@ end
     ]
     [@test dimension(addr) == dimension(typeof(addr)) for addr in addresses]
 end
+
+@testset "ExtendedHubbardReal1D_new_Boundary_conditions" begin
+    @testset "Gutzwiller" begin
```

Comment:
Why test Gutzwiller-specific things again here? It would be better to include tests that only check `ExtendedHubbardReal1D`.

Reply:
Thank you for your comments. I have did some of the changes. Please have a look again and let me know if any thing else I should do.